src/test/regress/sql/other_databases.sql

```diff
@@ -102,5 +102,23 @@ DROP USER IF EXISTS worker_user1, worker_user2, worker_user3;
 DROP DATABASE other_db2;
 \c - - - :master_port
 
+-- test creating and dropping a database from a Citus non-main database
+ALTER SYSTEM SET citus.enable_create_database_propagation TO true;
+SELECT pg_reload_conf();
+\c other_db1
+CREATE DATABASE other_db3;
+
+\c regression
+SELECT result FROM run_command_on_all_nodes($$SELECT datname FROM pg_database WHERE datname = 'other_db3'$$);
+
+\c other_db1
+DROP DATABASE other_db3;
+
+\c regression
+SELECT result FROM run_command_on_all_nodes($$SELECT datname FROM pg_database WHERE datname = 'other_db3'$$);
+
+ALTER SYSTEM SET citus.enable_create_database_propagation TO false;
+SELECT pg_reload_conf();
+
```

Comment:
Should we also add a test where CREATE DATABASE fails for some reason, just to see things don't break in case of a failure.

e.g., we can locally create a database X on one of the nodes and then try to create the same database X but this time in a distributed fashion from a non-maindb. And it would report an error while executing command on node-foo.

We can also test the same for DROP command too.